ci/test-sriov-secondary-network-aws.sh

```diff
@@ -0,0 +1,582 @@
+        echo -e "\tHostName $NODE_IP" >> "$SSH_CONFIG_FILE"
+        echo -e "\tPort 22" >> "$SSH_CONFIG_FILE"
+        echo -e "\tUser ubuntu" >> "$SSH_CONFIG_FILE"
+        echo -e "\tIdentityFile $THIS_DIR/../$AWS_EC2_SSH_KEY_NAME" >> "$SSH_CONFIG_FILE"
+    done
+
+    echo "SSH config written to $SSH_CONFIG_FILE"
+}
+
+function create_ippool_and_network_attachment_definition() {
+    subnet_info=$(aws ec2 describe-subnets --subnet-ids "$AWS_SUBNET_ID" --query 'Subnets[0].{CIDR:CidrBlock}' --output json)
+    subnet_cidr=$(echo "$subnet_info" | jq -r '.CIDR')
+
+    # Ensure valid CIDR is fetched
+    if [ "$subnet_cidr" == "null" ] || [ -z "$subnet_cidr" ]; then
+        echo "Error: Unable to fetch CIDR block for subnet $AWS_SUBNET_ID."
+        exit 1
+    fi
+
+    NETWORK=$(echo "$subnet_cidr" | cut -d/ -f1)
+    NETMASK=$(echo "$subnet_cidr" | cut -d/ -f2)
+    echo "Subnet CIDR network: $NETWORK"
+    echo "Subnet CIDR netmask: $NETMASK"
+
+    ip_pool_cidr="$(echo "$NETWORK" | awk -F'.' '{print $1 "." $2 "." $3 "." 192}')/26"
+    echo "Creating CIDR reservation $ip_pool_cidr in subnet $AWS_SUBNET_ID..."
+    aws ec2 create-subnet-cidr-reservation --subnet-id "$AWS_SUBNET_ID" --reservation-type explicit --cidr "$ip_pool_cidr"
+
+    SUBNET_CIDR_RES_ID=$(aws ec2 get-subnet-cidr-reservations --subnet-id "$AWS_SUBNET_ID" --query 'SubnetIpv4CidrReservations[0].SubnetCidrReservationId' --output text)
+    if [ "$SUBNET_CIDR_RES_ID" == "None" ]; then
+        echo "Error: Failed to create subnet CIDR reservation."
+        exit 1
+    fi
+
+    # Print the Subnet CIDR Reservation ID for reference
+    echo "CIDR reservation created with ID: $SUBNET_CIDR_RES_ID"
+
+    gateway=$(echo $NETWORK | awk -F'.' '{print $1 "." $2 "." $3 "." $4+1}')
+
+    # Create IP Pool YAML file
+    echo "Generating IPPool YAML..."
+    cat << EOF > "$IP_POOL_YAML"
+apiVersion: crd.antrea.io/v1beta1
+kind: IPPool
+metadata:
+  name: pool1
+spec:
+  ipRanges:
+  - cidr: $ip_pool_cidr
+  subnetInfo:
+    gateway: $gateway
+    prefixLength: $NETMASK
+EOF
+    echo "Created IP Pool YAML file: $IP_POOL_YAML"
+    cat $IP_POOL_YAML
+    kubectl apply -f $IP_POOL_YAML
+
+     # Create NetworkAttachmentDefinition
+    kubectl apply -f "$NAD_YAML"
+    echo "Created NetworkAttachmentDefinition"
+}
+
+function run_test() {
+     kubectl apply -f "$SRIOV_SECONDARY_NETWORKS_YAML"
+     kubectl apply -f https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/refs/heads/master/deployments/sriovdp-daemonset.yaml
+     kubectl get nodes -o go-template='{{range .items}}{{.metadata.name}}{{" "}}{{.status.allocatable}}{{"\n"}}{{end}}'
+     kubectl apply -f https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/raw/master/artifacts/networks-crd.yaml
+     create_ippool_and_network_attachment_definition
+     kubectl taint nodes --all node-role.kubernetes.io/control-plane- || true
```

Comment:
not sure this is actually needed: the `PodBuilder` takes care of it in the e2e test framework

https://github.com/antrea-io/antrea/blob/1143a884f9a2a2e23fc97d91d286cbd0031537cc/test/e2e/framework.go#L1575-L1578

Reply:
I have tried and without this line the test failed.https://jenkins.antrea.io/job/cloud-antrea-test-aws-sriov-secondary-network-copy/6/console